Userland/Libraries/LibArchive/Zip.cpp

```diff
@@ -138,6 +140,63 @@ ErrorOr<void> ZipOutputStream::add_member(ZipMember const& member)
     return local_file_header.write(*m_stream);
 }
 
+ErrorOr<double> ZipOutputStream::add_member_from_stream(StringView path, Stream& stream, Optional<Core::DateTime> optional_date)
+{
+    auto buffer = TRY(stream.read_until_eof());
+
+    Archive::ZipMember member {};
+    member.name = TRY(String::from_utf8(path));
+
+    if (optional_date.has_value()) {
+        auto date = optional_date.value();
+        member.modification_date = to_packed_dos_date(date.year(), date.month(), date.day());
+        member.modification_time = to_packed_dos_time(date.hour(), date.minute(), date.second());
+    }
+
+    auto deflate_buffer = Compress::DeflateCompressor::compress_all(buffer);
+    auto compression_ratio = 0.0;
+
+    if (!deflate_buffer.is_error() && deflate_buffer.value().size() < buffer.size()) {
+        member.compressed_data = deflate_buffer.value().bytes();
+        member.compression_method = Archive::ZipCompressionMethod::Deflate;
+
+        compression_ratio = (double)deflate_buffer.value().size() / buffer.size();
+    } else {
+        member.compressed_data = buffer.bytes();
+        member.compression_method = Archive::ZipCompressionMethod::Store;
+    }
+
+    member.uncompressed_size = buffer.size();
+
+    Crypto::Checksum::CRC32 checksum { buffer.bytes() };
+    member.crc32 = checksum.digest();
+    member.is_directory = false;
+
+    TRY(add_member(member));
+
+    return compression_ratio;
+}
+
+ErrorOr<void> ZipOutputStream::add_directory(StringView name, Optional<Core::DateTime> optional_date)
```

Comment:
`optional_date` does not describe what the argument represents. I'd name it something like `modification_time`. Same goes for `::add_member_from_stream`.